src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -525,6 +582,388 @@ public List<UserType> fetchUserTypes() {
+                .findByEmailId(profileRequest.getEmailId().toLowerCase());
+        if (caseWorkerProfile == null) {
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE,PROFILE_NOT_PRESENT_IN_DB,
+                    StringUtils.EMPTY,profileRequest,STAFF_PROFILE_UPDATE);
+            throw new ResourceNotFoundException(PROFILE_NOT_PRESENT_IN_DB);
+        }
+        return caseWorkerProfile;
+
+    }
+
+
+    public CaseWorkerProfile updateStaffProfiles(StaffProfileCreationRequest cwUiRequest,
+                                                       CaseWorkerProfile caseWorkerProfile) {
+
+        CaseWorkerProfile processedCwProfile;
+        Map<String, StaffProfileCreationRequest> emailToRequestMap = new HashMap<>();
+
+        //create map for input request to email
+        emailToRequestMap.put(cwUiRequest.getEmailId().toLowerCase(), cwUiRequest);
+
+        try {
+
+            //process update and suspend CW profiles
+            Pair<List<CaseWorkerProfile>, List<CaseWorkerProfile>> updateAndSuspendedLists = processExistingCaseWorkers(
+                    cwUiRequest, caseWorkerProfile);
+            // persist in db
+            CaseWorkerProfile cwProfileToPersist = null;
+            if (!ObjectUtils.isEmpty(updateAndSuspendedLists.getLeft())) {
+                cwProfileToPersist = updateAndSuspendedLists.getLeft().get(0);
+            }
+            if (!ObjectUtils.isEmpty(updateAndSuspendedLists.getRight())) {
+                cwProfileToPersist = updateAndSuspendedLists.getRight().get(0);
+            }
+
+            processedCwProfile = persistCaseWorker(cwProfileToPersist,
+                    emailToRequestMap);
+        } catch (Exception exp) {
+            log.error("{}:: createCaseWorkerUserProfiles failed  ::{}", loggingComponentName,
+                    exp);
+            throw exp;
+        }
+        return processedCwProfile;
+    }
+
+    public Pair<List<CaseWorkerProfile>, List<CaseWorkerProfile>> processExistingCaseWorkers(
+            StaffProfileCreationRequest cwUiRequest, CaseWorkerProfile caseWorkerProfiles) {
+
+        List<CaseWorkerProfile> updateCaseWorkerProfiles = new ArrayList<>();
+        List<CaseWorkerProfile> suspendedProfiles = new ArrayList<>();
+
+        if (isTrue(caseWorkerProfiles.getSuspended())) {
+            //when existing profile with delete flag is true then log exception add entry in exception table
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE, ALREADY_SUSPENDED_ERROR_MESSAGE,
+                    StringUtils.EMPTY, cwUiRequest, STAFF_PROFILE_UPDATE);
+        } else if (cwUiRequest.isSuspended()) {
+            //when existing profile with delete flag is true in request then suspend user
+            if (isUserSuspended(UserProfileUpdatedData.builder().idamStatus(IDAM_STATUS_SUSPENDED).build(),
+                    caseWorkerProfiles.getCaseWorkerId(), ORIGIN_EXUI, cwUiRequest.getRowId())) {
```

Comment:
I think you don't need `cwUiRequest.getRowId()` as this is not an excel upload

Reply:
done